src/Pmi/Order/Order.php

```diff
@@ -1352,4 +1352,85 @@ public function getReasonDisplayText()
+
+        if (!empty($object->identifier)) {
+            foreach ($object->identifier as $identifier) {
+                if (preg_match('/tracking-number/i', $identifier->system)) {
+                    $trackingNumber = $identifier->value;
+                }
+                if (preg_match('/PerfSite/i', $identifier->system)) {
+                    $performingSite = $identifier->value;
+                }
+                if (preg_match('/kit-id/i', $identifier->system)) {
+                    $kitId = $identifier->value;
+                }
+            }
+        }
+
+        // Extract participantId
+        preg_match('/^Patient\/(P\d+)$/i', $object->subject, $subject_matches);
+
+        $this->order['id'] = $object->id;
+        $this->order['participant_id'] = $subject_matches[1] ? $subject_matches[1] : 'Unknown';
+        $this->order['order_id'] = $kitId;
+        $this->order['rdr_id'] = $object->id;
+        $this->order['oh_type'] = 'kit';
+        $this->order['created_ts'] = $object->created;
+        $this->order['processed_ts'] = $processedTs;
+        $this->order['collected_ts'] = $collectedTs;
+        $this->order['finalized_ts'] = $finalizedTs;
+        $this->order['processed_centrifuge_type'] = (!empty($centrifugeType)) ? $centrifugeType : null;
+        $this->order['collected_samples'] = json_encode(!empty($collectedSamples) ? $collectedSamples : []);
+        $this->order['collected_ts'] = !empty($collectedTs) ? $collectedTs : null;
+        $this->order['processed_samples'] = json_encode(!empty($processedSamples) ? $processedSamples : []);
+        $this->order['processed_samples_ts'] = json_encode(!empty($processedSamplesTs) ? $processedSamplesTs : []);
+        $this->order['finalized_samples'] = json_encode(!empty($finalizedSamples) ? $finalizedSamples : []);
+        $this->order['finalized_ts'] = !empty($finalizedTs) ? $finalizedTs : null;
+        $this->order['collected_notes'] = $collectedNotes;
+        $this->order['processed_notes'] = $processedNotes;
+        $this->order['finalized_notes'] = $finalizedNotes;
+        $this->order['fedex_tracking'] = !empty($trackingNumber) ? $trackingNumber : null;
+        $this->order['collected_user_id'] = !empty($order->collectedInfo) ? $order->collectedInfo->author->value : false;
+        $this->order['collected_site'] = !empty($performingSite) ? $performingSite : null;
+        $this->order['processed_user_id'] = !empty($order->processedInfo) ? $order->processedInfo->author->value : false;
+        $this->order['processed_site'] = !empty($performingSite) ? $performingSite : null;
+        $this->order['finalized_user_id'] = !empty($order->finalizedInfo) ? $order->finalizedInfo->author->value : false;
+        $this->order['finalized_site'] = !empty($performingSite) ? $performingSite : null;
+        $this->order['failedToReachRDR'] = false;
+        $this->order['status'] = 'finalized';
+
```

Comment:
You should define these two fields to prevent undefined index notice on line 1056, 803 and 806. As this is kit order and since there are no custom sample selection you can set something like
`$this->order['type'] = 'kit';`
`$this->order['requested_samples']` = 'NULL';`

Reply:
Fixed in d2b89202